python/paddle/v2/__init__.py

```diff
@@ -85,22 +79,63 @@ def set_env(key, value):
         if envset is None:
             os.environ[key] = value
 
-    ht = os.popen("lscpu |grep \"per core\"|awk -F':' '{print $2}'|xargs")
-    ht = int(ht.read())
-    if ht == 1:  # ht is off
-        set_env("OMP_DYNAMIC", "false")
-        set_env("KMP_AFFINITY", "granularity=fine,compact,0,0")
-    else:
+    def is_hyperthreading_enabled():
+        '''If Hyper Threading is enabled.'''
+        if platform.system() == "Linux":
+            percore = os.popen(
+                "lscpu |grep \"per core\"|awk -F':' '{print $2}'|xargs")
+            return int(percore.read()) != 1
+        elif platform.system() == "Darwin":
+            physical = int(os.popen("sysctl hw.physicalcpu").read())
+            logical = int(os.popen("sysctl hw.logicalcpu").read())
+            return logical > physical
+        else:
+            # do not support on other platform yet
+            return False
+
+    def get_logical_processors():
+        '''Get the logical processors number'''
+        import platform
+        if platform.system() == "Linux":
+            processors = os.popen(
+                "grep \"processor\" /proc/cpuinfo|sort -u|wc -l")
+            return int(processors.read())
+        elif platform.system() == "Darwin":
+            processors = os.popen("sysctl hw.logicalcpu")
+            return int(processors.read())
+        else:
+            # do not support on other platform yet
+            return 1
```

Comment:
I think this function definition (from L82 to L108) can be significantly shortened while keeping the readability:

```python
def num_logical_processor():
    cmds = { "Linux" : "grep \"processor\" /proc/cpuinfo|sort -u|wc -l",
             "Darwin" : "sysctl hw.logicalcpu" }
    return int(os.popen(cmds.get(platform.system(), "expr 1")).read())
```

Reply:
Got it, Thx.